server/core/src/main/scala/sttp/tapir/server/interpreter/ServerInterpreter.scala

```diff
@@ -182,17 +184,26 @@ class ServerInterpreter[R, F[_], B, S](
   private def decodeBody[RAW, T](
       request: ServerRequest,
       values: DecodeBasicInputsResult.Values,
-      bodyInput: EndpointIO.Body[RAW, T]
+      bodyInput: EndpointIO.Body[RAW, T],
+      maxBodyLength: Option[Long]
   ): F[DecodeBasicInputsResult] = {
-    requestBody.toRaw(request, bodyInput.bodyType).flatMap { v =>
-      bodyInput.codec.decode(v.value) match {
-        case DecodeResult.Value(bodyV) => (values.setBodyInputValue(bodyV): DecodeBasicInputsResult).unit
-        case failure: DecodeResult.Failure =>
-          v.createdFiles
-            .foldLeft(monad.unit(()))((u, f) => u.flatMap(_ => deleteFile(f.file)))
-            .map(_ => DecodeBasicInputsResult.Failure(bodyInput, failure): DecodeBasicInputsResult)
+    requestBody
+      .toRaw(request, bodyInput.bodyType, maxBodyLength)
+      .flatMap { v =>
+        bodyInput.codec.decode(v.value) match {
+          case DecodeResult.Value(bodyV) => (values.setBodyInputValue(bodyV): DecodeBasicInputsResult).unit
+          case failure: DecodeResult.Failure =>
+            v.createdFiles
+              .foldLeft(monad.unit(()))((u, f) => u.flatMap(_ => deleteFile(f.file)))
+              .map(_ => DecodeBasicInputsResult.Failure(bodyInput, failure): DecodeBasicInputsResult)
+        }
+      }
+      .handleError {
```

Comment:
There are two ways this exception can be thrown. For streaming responses, `RequestBody` returns a `Stream` wrapped with `limitBytes()`, so it's a correct "value". The stream will start flowing only in the `serverLogic`, when user code runs it into an effect. If it fails with `StreamMaxLengthExceededException` in such case, it will be catched by `ExceptionHandler`.
Another case is this `toRaw` call, which sometimes runs a stream early in order to produce a raw value. If `F[DecodeBasicInputsResult]` returned by `toRaw` fails, the `ExceptionHandler` is not called. We get a HTTP 500, but I'm not sure where it's produced at. Maybe it's a bug and we should always handle failing `F` here in such a way that it goes to the `ExceptionHandler`?

Reply:
Ah, that's because `ExceptionInterceptor` is an `EndpointInterceptor`, so it's only called when an endpoint is successfully decoded (which doesn't happen when `toRaw` throws). If there's an exception during the decoding process, that's considered a bug and the fail-safe kicks in, defined separately in each interpreter, e.g. [here](https://github.com/softwaremill/tapir/blob/cee01ca89b0fac0985dd3ce5ebccfa436d4dd414/server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/NettyServerHandler.scala#L139).

Maybe the exception interceptor should be a `RequestInterceptor`? Then you wouldn't be able to have endpoint-specific exception handling, but maybe that's not useful anyway ... not sure :)

Is it possible to always return a `DecodeBasicInputsResult.Failure` in case the body is too long? That would allow avoiding throwing the exception in the first place

Reply:
> Is it possible to always return a DecodeBasicInputsResult.Failure in case the body is too long? That would allow avoiding throwing the exception in the first place
Only when you run streams in `toRaw`. If you just produce streams as values, it's considered as successful decoding, and the stream is passed to serverLogic. It will fail with an exception only when serverLogic decides to run it.

Reply:
Yeah the second scenario is clear. I think the problem here is with `toRaw` - but if we can simply return `DecodeBasicInputsResult.Failure` then no changes to `ExceptionInterceptor` would be needed?

Reply:
You mean make `toRaw` return a `RawValue(DecodeBasicInputsResult.Failure(...), Nil)`?

Reply:
Ah yes, that might be necessary. But I think it makes sense, when the body can't be decoded into a "raw" value?

Reply:
Actually, `RawValue` is parameterized, we need to return `RawValue[R](value: R)`. This `R` needs to match `RawBodyType[R]`, so we cannot put an error in it.

Reply:
But you can return a `DecodeResult[RawValue[R]]`?

Reply:
Ok, so you mean to change method signature, got it. Yep, why not.

Reply:
I spent a while changing `toRaw` to return a `F[DecodeResult[RawValue[R]]` and I'm not sure it's beneficial.
1. It allows removing `handleError` we had in `ServerInterpreter` in favor of a `flatMap` over `toRaw`, but we need to handle failing `Future`/`IO`/`ZIO` in all `ToRequestBody` implementations pretty much in the same manner, sth like:
```scala
  toRawValue[R](serverRequest, bodyType, maxBytes)
      .map((r: RawValue[R]) => DecodeResult.Value(r))
      .recover { 
        case e: StreamMaxLengthExceededException =>
          DecodeResult.Error("", e)
      }
```
repeated in each implementation.
2. We still need to handle the exception in `DefaultExceptionHandler` in case streaming fails during `serverLogic`

Reply:
I'm not sure I understand 1, but 2. is convincing - we won't escape from the fact that we might end up with a failed effect, which we want to handle in some special way. So let's keep consistency here.